packages/fxa-settings/src/pages/ResetPasswordWithRecoveryKeyVerified/index.test.tsx

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import ResetPasswordWithRecoveryKeyVerified from '.';
+import { logViewEvent, usePageViewEvent } from '../../lib/metrics';
+
+jest.mock('../../lib/metrics', () => ({
+  logViewEvent: jest.fn(),
+  usePageViewEvent: jest.fn(),
+}));
+
+describe('ResetPasswordWithRecoveryKeyVerified', () => {
+  it('renders default content as expected', () => {
+    render(<ResetPasswordWithRecoveryKeyVerified />);
+
+    const newAccountRecoveryKeyButton = screen.getByText(
+      'Generate a new account recovery key'
+    );
+    const continueToAccountLink = screen.getByText('Continue to my account');
+
+    expect(newAccountRecoveryKeyButton).toBeInTheDocument();
+    expect(continueToAccountLink).toBeInTheDocument();
+  });
+
+  it('emits the expected metrics when a user generate new recovery keys', async () => {
+    render(<ResetPasswordWithRecoveryKeyVerified />);
+    const newAccountRecoveryKeyButton = screen.getByText(
+      'Generate a new account recovery key'
+    );
+    fireEvent.click(newAccountRecoveryKeyButton);
+    expect(logViewEvent).toHaveBeenCalledWith(
```

Comment:
If https://github.com/mozilla/fxa/pull/14611/files#r1043725393, then this might be...

```
expect(logViewEvent).toHaveBeenCalledWith(
  'settings.reset-password-with-recovery-key-verified,
  'generate-new-key',
  {
    entrypoint_variation: 'react',
  }
);
```